go/carmen/memory_footprint.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package carmen
+
+import "github.com/Fantom-foundation/Carmen/go/common"
+
+type MemoryFootprint interface {
+	// Total provides the amount of bytes consumed by the database structure including all its subcomponents
+	Total() uint64
+
+	// String allow memory footprints to be used in format strings.
+	// It returns a breakdown of all MemoryFootprints within the tree.
+	String() string
```

Comment:
I would remove this. When using the` %s` pattern, `String() `is called implicitly, i.e. it does not have to be explicitly defined in the interface. And If I am not wrong, the convention is that this method is usually not explicitly defined in interfaces.  Furthermore, there is a build-in interface `Stringer` for this method, which can be uses at places where needed. This is the beauty of Golang that you can implement interface methods, but you do not have to explicitly say it.